src/dates/format-date.js

```diff
@@ -195,4 +195,32 @@ export default function formatDate(date, format, locale = "en") {
 
         return result;
     });
+}
+
+export function formatFormat(date, format, locale = "en") {
```

Comment:
Could we rename the function? As @tsvetomir said, we could use either 'dateFormatString', 'dateFormat' or even 'extractDateFormat'. Isn't the purpose of the function to 'extract' the full date format?

Also it will better if we place the function in a separate file. Another approach would be to rename the whole file, but this will be a-la breaking change. The common pieces between both methods could be put in a separate files too, e.g. formatters and the dateFormatRegExp.